### PR TITLE
feat: Persist node metadata and links

### PR DESCRIPTION
**Pull request by bjchambers**

### CODE REVIEW

File: .github/workflows/ci-unit-tests.yml [hunk 1]
```diff
@@ -90,11 +90,11 @@ jobs:
...
 
-#      - name: "Unit tests (knowledge-store)"
-#        if: ${{ needs.preconditions.outputs.libs_knowledge_store == 'true' }}
-#        env:
-#          OPENAI_API_KEY: "${{ secrets.E2E_TESTS_OPEN_AI_KEY }}"
-#        run: tox -e unit-tests -c libs/knowledge-store && rm -rf libs/knowledge-store/.tox
+      - name: "Unit tests (knowledge-store)"
+        if: ${{ needs.preconditions.outputs.libs_knowledge_store == 'true' }}
+        env:
+          OPENAI_API_KEY: "${{ secrets.E2E_TESTS_OPEN_AI_KEY }}"
```
Comment (cbornet) [inline .github/workflows/ci-unit-tests.yml:96]:
```suggestion
```

Comment (cbornet) [inline .github/workflows/ci-unit-tests.yml:96]:
```suggestion
```

File: libs/knowledge-store/ragstack_knowledge_store/graph_store.py [hunk 2]
```diff
@@ -46,14 +50,52 @@ class SetupMode(Enum):
     ASYNC = 2
     OFF = 3
 
+def _serialize_metadata(md: Dict[str, Any]) -> str:
+    print("Serializing MD")
+    if isinstance(md.get("links"), Set):
+        print("Copying")
```
Comment (cbornet) [inline libs/knowledge-store/ragstack_knowledge_store/graph_store.py:56]: remove print

File: libs/knowledge-store/ragstack_knowledge_store/graph_store.py [hunk 3]
```diff
@@ -46,14 +50,52 @@ class SetupMode(Enum):
...
+        md = md.copy()
+        print("Making list")
```
Comment (cbornet) [inline libs/knowledge-store/ragstack_knowledge_store/graph_store.py:58]: lots of print to remove

**Review by cbornet — CHANGES_REQUESTED** [review]
Some prints to remove.
Otherwise LGTM.

File: libs/knowledge-store/ragstack_knowledge_store/graph_store.py (hunk 2 above)
Comment (bjchambers) [reply libs/knowledge-store/ragstack_knowledge_store/graph_store.py:56]: Heh. Can you tell that initially, this was failing to serialize and just hanging?

**Review by cbornet — APPROVED** [review]

**Pull request merged**